external/mcp-servers/hf-translation-pr-generator/github_agent.py

```diff
@@ -0,0 +1,206 @@
+        except Exception as e:
+            return f"ERROR: Branch creation error: {str(e)}"
+
+    def _create_or_update_file_for_pr(self, file_path: str, content: str, branch_name: str) -> str:
+        """Create or update file in the branch."""
+        try:
+            user_repo = self.github_client.get_repo(f"{self.user_owner}/{self.user_repo}")
+            
+            commit_message = f"Add {file_path.split('/')[-2]} translation for {file_path.split('/')[-1]}"
+            
+            # Try to create file first
+            try:
+                user_repo.create_file(
+                    path=file_path,
+                    message=commit_message,
+                    content=content,
+                    branch=branch_name
+                )
+                return f"SUCCESS: File created - {file_path}"
+                
+            except GithubException as e:
+                if e.status == 422:
+                    # File exists, try to update
+                    try:
+                        existing_file = user_repo.get_contents(file_path, ref=branch_name)
+                        user_repo.update_file(
+                            path=file_path,
+                            message=commit_message,
+                            content=content,
+                            sha=existing_file.sha,
+                            branch=branch_name
+                        )
+                        return f"SUCCESS: File updated - {file_path}"
+                    except Exception as update_e:
+                        return f"ERROR: File update failed - {str(update_e)}"
+                else:
+                    raise e
+                    
+        except Exception as e:
+            return f"ERROR: File processing failed - {str(e)}"
+
+    def _create_pull_request_for_translation(self, title: str, body: str, head_branch: str, base_branch: str) -> str:
+        """Create pull request for translation."""
+        try:
+            base_repo = self.github_client.get_repo(f"{self.base_owner}/{self.base_repo}")
+            
+            # Format head for cross-repo PR
+            head = f"{self.user_owner}:{head_branch}"
+            
+            # Check for existing PR
+            existing_pr = self._check_existing_pr(base_repo, head, base_branch)
+            if existing_pr:
+                return f"ERROR: {existing_pr}"
+            
+            # Create PR
+            pr = base_repo.create_pull(
+                title=title,
+                body=body,
+                head=head,
+                base=base_branch
+            )
+            return f"PR creation successful: {pr.html_url}"
+            
+        except GithubException as e:
+            if e.status == 422:
+                error_msg = e.data.get("message", "Unknown error")
+                return f"ERROR: PR creation failed (422): {error_msg}"
+            return f"ERROR: PR creation failed: {str(e)}"
```

Comment:
```suggestion
        # except GithubException as e:
        #     if e.status == 422:
        #         error_msg = e.data.get("message", "Unknown error")
        #         return f"ERROR: PR creation failed (422): {error_msg}"
        #     return f"ERROR: PR creation failed: {str(e)}"
```
상단과 동일한 이유로 주석 처리해두었습니다!